From e61d4cbbc9c67416af83e812606b6ef7bcb79eab Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 25 Jun 2019 10:24:43 +0530
Subject: [PATCH] ninja backend: Quote args to compiler rules

These must be quoted in the same way as args in each compile line.
This is not needed in 0.50 anymore because we don't add compiler args
directly to the compiler rule there.

This is needed to correctly quote arguments added through c_args in
the cross file.
---
 mesonbuild/backend/ninjabackend.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/backend/ninjabackend.py b/mesonbuild/backend/ninjabackend.py
index f49649be..de178b38 100644
--- a/mesonbuild/backend/ninjabackend.py
+++ b/mesonbuild/backend/ninjabackend.py
@@ -1599,7 +1599,7 @@ rule FORTRAN_DEP_HACK%s
             command_template = ' command = {executable} $ARGS {cross_args} {dep_args} {output_args} {compile_only_args} $in\n'
         command = command_template.format(
             executable=' '.join([ninja_quote(i) for i in compiler.get_exelist()]),
-            cross_args=' '.join(cross_args),
+            cross_args=' '.join([quote_func(i) for i in cross_args]),
             dep_args=' '.join(quoted_depargs),
             output_args=' '.join(compiler.get_output_args('$out')),
             compile_only_args=' '.join(compiler.get_compile_only_args())
-- 
2.21.0

